tests/framework/prototype.py

```diff
@@ -244,6 +245,37 @@ def assert_rule(self, custom_resource=None, cr_namespace='default'):
         pod_exec_command(headpod_name, cr_namespace,
             "python -c \"import ray; ray.init(); print(ray.cluster_resources())\"")
 
+class ShutdownJobRule(Rule):
+    """Check the Ray cluster is shutdown when setting `spec.shutdownAfterJobFinishes` to true."""
+    def assert_rule(self, custom_resource=None, cr_namespace='default'):
+        custom_api = client.CustomObjectsApi()
+        # Wait for there to be no RayClusters
+        logger.info("Waiting for RayCluster to be deleted...")
+        for i in range(30):
+            rayclusters = custom_api.list_namespaced_custom_object(
+                group = 'ray.io', version = 'v1alpha1', namespace = cr_namespace,
+                plural = 'rayclusters')
+            # print debug log
+            if i != 0:
+                logger.info("ShutdownJobRule wait() hasn't converged yet.")
+                logger.info("Number of RayClusters: %d", len(rayclusters["items"]))
+            if len(rayclusters["items"]) == 0:
+                break
+            time.sleep(1)
+        else:
+            raise TimeoutError("RayCluster hasn't been deleted in 30 seconds.")
+
+        logger.info("RayCluster has been deleted.")
+        
+
+
+    def trigger_condition(self, custom_resource=None) -> bool:
+        # Trigger if shutdownAfterJobFinishes is set to true
+        steps = "spec.shutdownAfterJobFinishes".split('.')
+        value = search_path(custom_resource, steps)
+        logger.info("ShutdownJobRule trigger_condition(): %s", value)
+        return value is not None and value
```

Comment:
`value` seems to be a string here. In my understanding, `value is not None and value` will return a string.

<img width="282" alt="Screen Shot 2023-08-01 at 11 17 12 AM" src="https://github.com/ray-project/kuberay/assets/20109646/4027b5fa-9747-4b87-b377-cb2b0bf08711">

Reply:
Oh, great catch!

Reply:
Actually `value` seems to be a `bool`.  I added an assertion just to be safe: https://github.com/ray-project/kuberay/pull/1269/files#diff-665dbdd284a6d0222ec3ddb2710957546d6c3974808765d71fb73d119d4cbef8R277
```
assert isinstance(value, bool) or value is None
```